superset/assets/src/dashboard/components/FilterIndicatorsContainer.jsx

```diff
@@ -0,0 +1,162 @@
+  getFilterIndicators() {
+    const {
+      dashboardFilters,
+      chartId: currentChartId,
+      filter_immune_slices,
+      filter_immune_slice_fields,
+    } = this.props;
+
+    if (Object.keys(dashboardFilters).length === 0) {
+      return [];
+    }
+
+    const dashboardFiltersColorMap = getFilterColorMap();
+
+    const sorteIndicatorsByEmptiness = Object.values(dashboardFilters).reduce(
+      (indicators, dashboardFilter) => {
+        const {
+          chartId,
+          componentId,
+          directPathToFilter,
+          scope,
+          isDateFilter,
+          isInstantFilter,
+          columns,
+        } = dashboardFilter;
+
+        // do not apply filter on filter_box itself
+        // do not apply filter on filter_immune_slices list
+        if (
+          currentChartId !== chartId &&
+          !filter_immune_slices.includes(currentChartId)
+        ) {
+          Object.keys(columns).forEach(name => {
+            const colorMapKey = getFilterColorKey(chartId, name);
+            const directPathToLabel = directPathToFilter.slice();
+            directPathToLabel.push(`LABEL-${name}`);
+            const indicator = {
+              chartId,
+              colorCode: dashboardFiltersColorMap[colorMapKey],
+              componentId,
+              directPathToFilter: directPathToLabel,
+              scope,
+              isDateFilter,
+              isInstantFilter,
+              name,
+              values: columns[name],
+            };
+
+            // do not apply filter on fields in the filter_immune_slice_fields map
+            if (
+              filter_immune_slice_fields[currentChartId] &&
+              filter_immune_slice_fields[currentChartId].includes(name)
+            ) {
+              return;
+            }
+
+            if (isEmptyValue(indicator.values)) {
+              indicators[1].push(indicator);
+            } else {
+              indicators[0].push(indicator);
+            }
+          });
+        }
+
+        return indicators;
+      },
+      [[], []],
+    );
+
+    // cypress' electron don't support [].flat():
+    return [...sorteIndicatorsByEmptiness[0], ...sorteIndicatorsByEmptiness[1]];
+  }
+
+  render() {
+    const { chartStatus, setDirectPathToChild } = this.props;
+    if (chartStatus === 'loading') {
+      return null;
+    }
+
+    const indicators = this.getFilterIndicators();
+    // if total indicators <= 5, show all
+    // else: show top 4 indicators, and show all the rest in group
+    const showExtraIndicatorsInGroup =
+      indicators.length > FILTER_INDICATORS_DISPLAY_LENGTH + 1;
+
+    return (
+      <div className="dashboard-filter-indicators-container">
+        {indicators
+          .filter((indicator, index) =>
+            showExtraIndicatorsInGroup
+              ? index < FILTER_INDICATORS_DISPLAY_LENGTH
+              : true,
+          )
+          .map((indicator, index) => (
+            <FilterIndicator
+              key={index}
```

Comment:
`indicator.name` instead?

Reply:
same as the other place, now i use ${indicator.chartId}_${indicator.name} as key.